### DISCUSSION TITLE
Run a Recipe Before Every Recipe

### DISCUSSION THREAD

**Discussion by hisaac** [casey/just#988] [Q&A] [answered]
Is it possible to have a kind of setup recipe that gets run at the start of every `just` command?

**Comment by casey** [comment, marked as answer]
Not currently. Can you define a setup recipe, and then make it a dependency of every other recipe? Like so:

```
setup:
  # do some setup

foo: setup
  # do foo stuff

bar: setup
  # do bar stuff
```

**Reply by hisaac** [reply]
Yep, that's what I'll do for now, thanks!
